### PR TITLE
Add feature of uploading files and creating a new folder

### PR DESCRIPTION
**Pull request by kuzmavictor**
During this iteration was implemented:
- the functionality  that allows creating a new folder;
- the handling error when the process creating folder will be failed.

### CODE REVIEW

**Comment by kuzmavictor** [comment]
@dkalinin PTAL.

File: app/view/component/page/file-list/index.js [hunk 1]
```diff
@@ -168,4 +176,45 @@ export class FileListPage extends ConfigComponent {
...
+    this._filesUploadComponent.addClassForLabel('upload-files-label');
+
+    this._filesUploadComponent.onClick(() => {
+      const files = this._filesUploadComponent.getInputFiles();
+      console.log(files);
+      this.dispatch('addFiles', Array.from(files), this._currentFolder.id);
+    });
+    this._createFolderButton = new Button(this._wrapperListButton, 'Create folder');
+    this._createFolderButton.onClick(() => {
+      this._fileList.controlNewFolderComponent();
```
Comment (dkalinin) [inline app/view/component/page/file-list/index.js]: _fileList should have the following API to handle new folder creation:
onFolderCreationCompleted()
onFolderCreationCancelled()
property: newFolderModel;

Action like 'createTemporaryFolder' adds new model to 'temporaryFolder' field of state.

**Comment by kuzmavictor** [comment]
@AndriiSameliuk @dkalinin fixed, PTAL.

File: app/view/component/page/file-list/index.js [hunk 2]
```diff
@@ -67,6 +73,57 @@ export class FileListPage extends ConfigComponent {
...
+   *
+   * @param {String} defaultNewFolderName - a default name to new folder
+   * @see Store
+   * @private
+   */
+  _changeCreatingFolderButtonName(defaultNewFolderName) {
+    if (defaultNewFolderName) {
+      this._createFolderButton.text = CLOSE_CREATING_FOLDER_BTN;
+    } else {
+      this._createFolderButton.text = CREATING_FOLDER_BTN;
```
Comment (dkalinin) [inline app/view/component/page/file-list/index.js]:
```js 
this._createFolderButton.text = defaultNewFolderName ? CLOSE_CREATING_FOLDER_BTN : CREATING_FOLDER_BTN;
```

Comment (dkalinin) [reply app/view/component/page/file-list/index.js]: CREATING_FOLDER_BTN => CREATING_FOLDER_TEXT
CLOSE_CREATING_FOLDER_BTN => CANCEL_CREATING_FOLDER_TEXT

File: app/service/store/action/index.js [hunk 3]
```diff
@@ -68,6 +69,7 @@ export const actions = {
     return store.requestServiceProvider.fileListService.getFiles(store.state.currentFolder.id)
         .then((fileList) => {
           store.commit('setFiles', fileList);
+          store.commit('setTemporaryFolderName', undefined);
```
Comment (dkalinin) [inline app/service/store/action/index.js]: Please remove. It should be done in createFolder action.

**Pull request merged**